cli/lsp/language_server.rs

```diff
@@ -2782,6 +2889,204 @@ mod tests {
     harness.run().await;
   }
 
+  #[tokio::test]
+  async fn test_semantic_tokens() {
+    let mut harness = LspTestHarness::new(vec![
+      ("initialize_request.json", LspResponse::RequestAny),
+      ("initialized_notification.json", LspResponse::None),
+      (
+        "semantic_tokens_did_open_notification.json",
+        LspResponse::None,
+      ),
+      (
+        "semantic_tokens_full_request.json",
+        LspResponse::Request(
+          2,
+          json!({
+            "data": [
+              0,
+              5,
+              6,
+              1,
+              1,
+              0,
+              9,
+              6,
+              8,
+              9,
+              0,
+              8,
+              6,
+              8,
+              9,
+              2,
+              15,
+              3,
+              10,
+              5,
+              0,
+              4,
+              1,
+              6,
+              1,
+              0,
+              12,
+              7,
+              2,
+              16,
+              1,
+              8,
+              1,
+              7,
+              41,
+              0,
+              4,
+              1,
+              6,
+              0,
+              0,
+              2,
+              5,
+              11,
+              16,
+              1,
+              9,
+              1,
+              7,
+              40,
+              3,
+              10,
+              4,
+              2,
+              1,
+              1,
+              11,
+              1,
+              9,
+              9,
+              1,
+              2,
+              3,
+              11,
+              1,
+              3,
+              6,
+              3,
+              0,
+              1,
+              0,
+              15,
+              4,
+              2,
+              0,
+              1,
+              30,
+              1,
+              6,
+              9,
+              1,
+              2,
+              3,
+              11,
+              1,
+              1,
+              9,
+              9,
+              9,
+              3,
+              0,
+              16,
+              3,
+              0,
+              0,
+              1,
+              17,
+              12,
+              11,
+              3,
+              0,
+              24,
+              3,
+              0,
+              0,
+              0,
+              4,
+              9,
+              9,
+              2
+            ]
```

Comment:
I would try to use `#[rustfmt::skip]` here and reformat this in some more readable way manually, or externalise the JSON file, because this is just a lot of useless vertical space in an already big file.

Reply:
done